### ISSUE TITLE
Remove SDL??

### ISSUE THREAD

**Issue by dressupgeekout** [dressupgeekout/lugarudc#9]
Lugaru is written against SDL2 but there's a port of SDL1 for KallistiOS. Lugaru's utilization of SDL is pretty tame, therefore it shouldn't be too difficult to bring the game back to SDL1 if needed. Lugaru was originally written for SDL 1.2, basically we would need to revert 1c7d28a2fc93699ab955ea66203d0c36bba9d977

HOWEVER, I think we might be able to do away with SDL altogether and save some RAM? Quickly grepping for `SDL_*` in Lugaru reveals the following:

- Lots of PC-only considerations, e.g. "fullscreen or windowed?"
- A crapton of keyboard handling, which won't be necessary because our port won't have keyboard support
- Some mouse handling, which won't be necessary because our port won't have mouse support

Most notably, I DON'T see SDL being used to draw anything to the screen. It's true that we do use SDL to create a GL context and all that, but we never actually `SDL_RendererFlip()` or `SDL_BlitSurface()` or anything like that. Instead Lugaru does a bunch of `glThis()` and `glThat()` directly. That's why I think we can nix SDL entirely and use KallistiOS's input handling instead.

**Comment by dressupgeekout** [comment]
Yeah, the `SDL_Window` type isn't even available in the KOS port of SDL 1.2 sooo I say let's just forget about it.

**Comment by dressupgeekout** [comment]
Ugh, but maybe it makes sense to keep SDL to allow the game to be iterated upon and testable on your development machine. Decisions decisions...

Here's what I'm currently thinking:

- The project would support 2 configurations: "Dreamcast mode" and "development mode"
- Dreamcast-mode would NOT leverage SDL, despite the fact a port of SDL 1.2 for KOS already exists
- Development mode WILL leverage SDL2 as it currently does
- Lots of `#ifdef DREAMCAST` and `#ifndef DREAMCAST` would be sprinkled throughout the input-handling and screen-drawing code

**Comment by gyrovorbis** [comment]
Ughghghg. You know, we take great pride in having great ports of the most common libraries and APIs, and we're always adding stuff to kos-ports...

...but our lack of SDL2 is the single most disappointing thing for me in terms of what our SDK has to offer. Our SDL1 support was so good that we even have OpenGL interoperability. It was so good that the SDL guys came to our repo during SDL2's development and were like "yo dawg, you guys ever going to do an SDL2 port!?" But unfortunately nobody ever stepped up to do it. :(

This was before my time working on KOS, but I've always sworn that I'd eventually get around to at least starting the port... I've even reached back out to the SDL guys to see if SDL3 could even be possible on DC, and it still seems like it should be.

Anyway, sorry to bore you with this stuff. This looks really cool. Feel free to stop by the Simulant discord server where most of the DC developers (including us KOS devs) typically hang out!

**Comment by dressupgeekout** [comment]
@gyrovorbis Yeah I've definitely hovered back and forth over whether I should just take the plunge and see what it really takes to bring SDL2 to KOS... I mean, the appetite is definitely there. Regardless, this Lugaru project doesn't need to be blocked on that. Although it would be REALLY COOL to have everything Just Work out of the box.

I've been lurking in the Simulant Discord for a long time now, heh
